exporters/provider_common/openshift.py

```diff
@@ -1,11 +1,60 @@
+import logging
+import re
+import time
 from datetime import datetime
-from typing import Union
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from openshift.dynamic import DynamicClient, ResourceInstance
+from openshift.dynamic.exceptions import ResourceNotFoundError
+from openshift.dynamic.resource import ResourceField
 
 from pelorus.timeutil import parse_assuming_utc
 
 # https://docs.openshift.com/container-platform/4.10/rest_api/objects/index.html#io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta
 _DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
 
+SUPPORTED_REPLICA_OBJECTS = ["ReplicaSet", "ReplicationController"]
+
+# Cache threshold in seconds, used by every cached_parents_dict entry
+CACHE_THRESHOLD_1_DAY = 60 * 60 * 24
+cached_parents_dict: dict[str, Tuple[ResourceInstance, float]] = {}
+
+
+def _add_object_to_cache(uid: str, k8s_obj: ResourceInstance) -> None:
+    """
+    Create in-memory cache for the K8S objects, so we don't have
+    to query them each time.
+
+    We have also 'timeout' for each cache entry, which means
+    we won't grow the cache infinitely.
+    """
+    if uid not in cached_parents_dict:
+        cached_parents_dict[uid] = (k8s_obj, time.time())
+
+
+def _get_object_from_cache(uid: str) -> ResourceInstance:
+    """
+    Gets the object from the cache by it's uid.
+    """
+    k8s_obj, insertion_time = cached_parents_dict.get(uid) or (None, None)
```

Comment:
here there is an unused variable (and in the remove function). We can use `_` in these cases

example
```py
k8s_obj, _ = cached_parents_dict.get(uid, (None, None))
return k8s_obj
```